specviz/third_party/glue/viewer.py

```diff
@@ -159,23 +183,34 @@ def update(self, *args, **kwargs):
             return
 
         try:
-            spectrum = glue_data_to_spectrum1d(self.state.layer, self.state.attribute, statistic=self.state.statistic)
+            spectrum = glue_data_to_spectrum1d(self.state.layer,
+                                               self.state.attribute,
+                                               statistic=self.state.statistic)
         except IncompatibleAttribute:
             self.disable_invalid_attributes(self.state.attribute)
             return
 
         self.enable()
 
         if self.data_item is None:
-            self.data_item = self.specviz_window.model.add_data(spectrum, name=self.state.layer.label)
-            self.plot_widget.add_plot(self.plot_data_item, visible=True, initialize=True)
+            self.data_item = self.specviz_window.model.add_data(spectrum,
+                                                                name=self.state.layer.label)
+            self.plot_widget.add_plot(self.plot_data_item,
+                                      visible=True,
+                                      initialize=True)
         else:
             self.plot_data_item.data_item.set_data(spectrum)
             # FIXME: we shouldn't have to call update_data manually
```

Comment:
Remove out of date comment? (or update)

Reply:
It's still relevant. It's a "workaround" right now -- we do _want_ it to update automatically, but hooking up requires a bit of a refactor.